environs/interface.go

```diff
@@ -339,16 +339,24 @@ type BootstrapEnviron interface {
 	// StorageProviders returned from Environ.StorageProvider will
 	// be scoped specifically to that Environ.
 	storage.ProviderRegistry
+}
+
+// ModelResources provides the API to use when creating new models.
+type ModelResources interface {
+	// ValidateModelCreation returns an error if a new model
+	// cannot be created owing to issues such as incompatible or
+	// otherwise invalid model configuration.
+	ValidateModelCreation(context.Context) error
```

Comment:
This is validating that the provider is in a fit enough state to host model x. I think the name needs to be better reflect this.

Reply:
renamed to `ValidateProviderForNewModel`